### PR TITLE
Use a tempdir in smoketest when downloading.

### PR DESCRIPTION
**Pull request by DailyDreaming**

### CODE REVIEW

File: tests/infra/base_smoketest.py [hunk 1]
```diff
@@ -183,7 +183,14 @@ def _test_get_subscriptions(self, replica, requested_subscription):
         list_of_subscription_uuids = [x['uuid'] for x in list_of_subscriptions if x['uuid']]
         self.assertIn(requested_subscription, list_of_subscription_uuids)
 
-    def _test_replica_sync(self, current_replica, bundle_uuid, ):
+    def _download_bundle(self, replica_name: str, bundle_uuid: str):
+        tempdir = tempfile.mkdtemp()
```
Comment (amarjandu) [inline tests/infra/base_smoketest.py]: can the tempdir take the `cls.workdir` into context, and keep everything contained inside?

Comment (DailyDreaming) [reply tests/infra/base_smoketest.py]: I suppose so.  I'll add it in.

Comment (xbrianh) [reply tests/infra/base_smoketest.py]: Agreeing with Amar:
```
with tempfile.TemporaryDirectory() as tempdir:
    pass
```

From the [documentation](https://docs.python.org/3.6/library/tempfile.html#tempfile.TemporaryDirectory):
> On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.

File: tests/infra/base_smoketest.py [hunk 2]
```diff
@@ -183,7 +183,14 @@ def _test_get_subscriptions(self, replica, requested_subscription):
         list_of_subscription_uuids = [x['uuid'] for x in list_of_subscriptions if x['uuid']]
         self.assertIn(requested_subscription, list_of_subscription_uuids)
 
-    def _test_replica_sync(self, current_replica, bundle_uuid, ):
+    def _download_bundle(self, replica_name: str, bundle_uuid: str):
+        tempdir = tempfile.mkdtemp(prefix=f'{cls.workdir.name}/')
```
Comment (xbrianh) [inline tests/infra/base_smoketest.py]: Agreeing with Amar:
```
with tempfile.TemporaryDirectory() as tempdir:
    pass
```

From the [documentation](https://docs.python.org/3.6/library/tempfile.html#tempfile.TemporaryDirectory):
> On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.

Comment (DailyDreaming) [reply tests/infra/base_smoketest.py]: Addressed.

File: tests/infra/base_smoketest.py [hunk 3]
```diff
@@ -183,7 +183,12 @@ def _test_get_subscriptions(self, replica, requested_subscription):
         list_of_subscription_uuids = [x['uuid'] for x in list_of_subscriptions if x['uuid']]
         self.assertIn(requested_subscription, list_of_subscription_uuids)
 
-    def _test_replica_sync(self, current_replica, bundle_uuid, ):
+    def _download_bundle(self, replica_name: str, bundle_uuid: str):
+        with tempfile.TemporaryDirectory(prefix=f'{self.workdir.name}/') as tempdir:  # type: ignore
+            run(f"{self.venv_bin}hca dss download --replica {replica_name} --bundle-uuid {bundle_uuid}"  # type: ignore
```
Comment (xbrianh) [inline tests/infra/base_smoketest.py]: Can we get rid of the `type: ignore`s?

Comment (DailyDreaming) [reply tests/infra/base_smoketest.py]: MyPy says that `self.workdir` and `self.venv_bin` don't exist.

**Pull request merged**